### PR TITLE
[Cody GA Launch] Improvements to Cody main overview docs page

### PR DESCRIPTION
**Pull request by MaedahBatool**
This PR ships improvement to the main Cody overview page. Includes changes like:

- Onboarding video for quick product overview
- Details about data collection and usage
- Details about compatibility with Sourcegraph products

## Test plan

Docs added

## Preview 🤩
[Preview Link](https://docs.sourcegraph.com/@imp-cody-home)

### CODE REVIEW

File: doc/cody/overview/index.md [hunk 1]
```diff
@@ -92,57 +92,79 @@ td:first-child {
...
+Cody is a free and open-source AI coding assistant that writes, fixes, and maintains your code. Cody understands your entire codebase by leveraging the power of [Code Graph](./../core-concepts/code-graph.md) to gather context, which assists you in writing accurate code.
 
-Think of Cody as your personal AI coding assistant, equipped with a comprehensive understanding of three crucial elements:
+<video width="1920" height="1080" loop playsinline controls style="width: 100%; height: auto; max-width: 50rem">
+  <source src="https://storage.googleapis.com/sourcegraph-assets/Docs/Media/1.mp4" type="video/mp4">
+</video>
+
+Cody connects seamlessly with codehosts like <a target="blank" href="https://sourcegraph.com/get-cody">GitHub</a>, <a target="blank" href="https://gitlab.com/users/sign_in">GitLab</a> and IDEs like <a target="blank" href="https://marketplace.visualstudio.com/items?itemName=sourcegraph.cody-ai">VS Code</a> and <a target="blank" href="https://plugins.jetbrains.com/plugin/9682-sourcegraph-cody--code-search">JetBrains</a>. Once connected, Cody acts as your personal AI coding assistant, equipped with a comprehensive understanding of the following three crucial elements:
```
Comment (jdorfman) [inline doc/cody/overview/index.md:135]: GitHub links to https://sourcegraph.com/get-cody not sure if that is intentional.

File: doc/cody/overview/index.md [hunk 2]
```diff
@@ -163,22 +185,27 @@ Cody's main features include:
...
 
-- [Cody clients, plugins, and extensions](./../core-concepts/cody_clients.md)
-- [Enabling Cody for Sourcegraph Enterprise customers](./../overview/enable-cody-enterprise.md)
-- [Enabling Cody for the Cody app](../overview/app/index.md)
-- [Enabling Cody for open source Sourcegraph.com users](./../overview/cody-with-sourcegraph.md)
-- [Installing the Cody VS Code extension](./../overview/install-vscode.md)
-- [Installing the Jetbrains extension (experimental)](./../overview/install-jetbrains.md)
-- [Configuring code graph context](./../core-concepts/code-graph.md)
-- [Sourcegraph Cody Gateway](./../core-concepts/cody_gateway.md)
+If you have any questions regarding Cody, you can always ask our community on [GitHub discussions](https://github.com/sourcegraph/cody/discussions), [Discord](https://discord.com/invite/s2qDtYGnAE), or [create a post on X](https://twitter.com/sourcegraph).
```
Comment (jdorfman) [inline doc/cody/overview/index.md:208]: https://x.com/sourcegraphcody

**Review by jdorfman — CHANGES_REQUESTED** [review]

File: doc/cody/overview/index.md (hunk 1 above)
Comment (MaedahBatool) [reply doc/cody/overview/index.md:135]: I have asked Alex about it. This is how it appears on the about website.

File: doc/cody/overview/index.md (hunk 2 above)
Comment (MaedahBatool) [reply doc/cody/overview/index.md:209]: Oh nice catch.

**Review by jdorfman — APPROVED** [review]

**Pull request merged**